fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/RecordQueryPlanner.java

```diff
@@ -470,6 +466,23 @@ private ScoredPlan planFilterWithInJoin(@Nonnull PlanContext planContext, @Nonnu
         return null;
     }
 
+    // Get the key expression for the index entries, which includes primary key fields for normal indexes.
+    private KeyExpression indexKeyExpressionForPlan(@Nullable KeyExpression commonPrimaryKey, @Nonnull Index index) {
```

Comment:
Is this only ever called for value indexes? if so, let's put a `Verify.verify()` in here!

Reply:
It is called for every index that the planner considers and needs to return something always. The specific handling of value index types is a few lines down (`indexTypes.getValueTypes().contains(index.getType())`), in which case it just returns the as-declared key.